lib/field.ml

```diff
@@ -124,6 +126,11 @@ let time field =
   | `Time t -> t
   | _ -> err field ~info:"a time value"
 
+let json field =
+  match value field with
+  | `Json j -> j
+  | _ -> err field ~info:"a json value"
+
```

Comment:
In the test case you take into account the possibility that the database server returns a string (or bytes, as I've when testing).  That would be needed here, as well.  But if you agree to the above, this definition can be omitted.